### ISSUE TITLE
Ease Spreadoperator in particular cases

### ISSUE THREAD

**Issue by vanniktech** [detekt/detekt#1045]
Sometimes when interacting with certain APIs you're forced to use the SpreadOperator. At least I haven't found a way around it. Such an example is the following:

Calling the following java function:

```java
public QueryObservable createQuery(@NonNull final Iterable<String> tables, @NonNull String sql,
      @NonNull String... args)
```

as 

```kotlin
briteDatabase.createQuery(it.tables, it.statement, *it.args)
```

Where `it` is the following java class:

```java
public class SqlDelightStatement {
  public final String statement;
  public final String[] args;
  /** A set of the tables this statement observes. */
  public final Set<String> tables;

  public SqlDelightStatement(String statement, String[] args, Set<String> tables) {
    this.statement = statement;
    this.args = args;
    this.tables = tables;
  }
}
```

Probably type resolution would be needed here to find such occurrences.

**Comment by schalkms** [comment]
I think this is related to issue #391 and also a duplicate.

**Comment by vanniktech** [comment]
Seems related but not a duplicate in my opinion.

**Comment by vanniktech** [comment]
Would be better if this would just work out of the box - but I guess that requires type resolution.

**Comment by BraisGabin** [comment]
I think that this was fixed at #3145. Please, add a comment if this issue wasn't addressed yet and we can reopen it.
